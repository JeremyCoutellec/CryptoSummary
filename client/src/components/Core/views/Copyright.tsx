import React from 'react';
import { Link } from 'react-router-dom';

// Material UI
import Typography from '@mui/material/Typography';

const Copyright = () => (
    <Typography variant="caption" color="inherit" align="center">
        {'Â©  2021 '}
        <Link className="color-white" to="#">
            CryptoSummary
        </Link>
        {` v${process.env.REACT_APP_VERSION}`}
    </Typography>
);

export default Copyright;
