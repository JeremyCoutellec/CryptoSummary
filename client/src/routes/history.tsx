import { createBrowserHistory } from 'history';

// History use for the browser navigation
export default createBrowserHistory();
