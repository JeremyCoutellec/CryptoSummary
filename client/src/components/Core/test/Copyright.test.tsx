import React from 'react';
import { createBrowserHistory } from 'history';
import { Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Copyright from '../views/Copyright';

test('Copyright rendering', () => {
    const propsCopyright = {};
    const history = createBrowserHistory();
    const component = renderer.create(
        <Router history={history}>
            <Copyright {...propsCopyright} />
        </Router>
    );
    const copyright = component.toJSON();
    expect(copyright).toMatchSnapshot();
});
