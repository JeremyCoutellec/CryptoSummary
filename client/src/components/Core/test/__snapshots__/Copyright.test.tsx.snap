// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Copyright rendering 1`] = `
<span
  className="MuiTypography-root MuiTypography-caption MuiTypography-alignCenter css-1228yko-MuiTypography-root"
>
  Â©  2021 
  <a
    className="color-white"
    href="/"
    onClick={[Function]}
  >
    CryptoSummary
  </a>
   v0.4.6.2
</span>
`;
